src/Microsoft.NetCore.Analyzers/Core/Runtime/InstantiateArgumentExceptionsCorrectly.Fixer.cs

```diff
@@ -13,17 +21,125 @@ public abstract class InstantiateArgumentExceptionsCorrectlyFixer : CodeFixProvi
+                    break;
+                case "IncorrectMessage":
+                    break;
+                case "IncorrectParameterName":
+                    break;
+            }
+
+            Task RegisterNoArgumentsCodeFixesAsync()
+            {
+                var constructors = GetCandidateReplacementConstructors(objectCreation, semanticModel);
+
+                foreach (IMethodSymbol constructor in constructors)
+                {
+                    string title = string.Format(
+                        SystemRuntimeAnalyzersResources.InstantiateArgumentExceptionsCorrectlyTitleNoArguments,
+                        constructor.ToMinimalDisplayString(semanticModel, objectCreation.SpanStart));
+                    context.RegisterCodeFix(
+                        CodeAction.Create(
+                            title,
+                            async ct => await ChangeConstructorInvocationAsync(document, objectCreation, constructor, cancellationToken).ConfigureAwait(false),
+                            equivalenceKey: title),
+                        context.Diagnostics);
+                }
+
+                return Task.CompletedTask;
+            }
         }
 
-        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        #region No arguments
+
+        private static IEnumerable<IMethodSymbol> GetCandidateReplacementConstructors(
+            SyntaxNode objectCreation,
+            SemanticModel semanticModel)
         {
-            // Fixer not yet implemented.
-            return Task.CompletedTask;
+            var emptyConstructor = (IMethodSymbol)semanticModel.GetSymbolInfo(objectCreation).Symbol;
+            Debug.Assert(emptyConstructor.Parameters.IsEmpty);
+
+            var exceptionType = emptyConstructor.ContainingType;
+            foreach (IMethodSymbol constructor in exceptionType.Constructors)
+            {
+                if (constructor == emptyConstructor)
+                {
+                    continue;
+                }
 
+                var parameters = constructor.Parameters;
+                switch (parameters.Length)
+                {
+                    case 1:
+                        if (IsMessage(parameters[0]) || IsParameterName(parameters[0]))
+                        {
+                            yield return constructor;
+                        }
+                        break;
+                    case 2:
+                        if ((IsMessage(parameters[0]) && IsParameterName(parameters[1])) ||
+                            (IsParameterName(parameters[0]) && IsMessage(parameters[1])))
+                        {
+                            yield return constructor;
+                        }
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+
+        private static bool IsMessage(IParameterSymbol parameter)
```

Comment:
@333fred I copied these methods from the analyzer file. Is there a convention used to share code between analyzers/fixers?

Reply:
If they're common enough to warrant sharing between multiple analyzers, then we put them in Analyzers.Utilities. If you just want to share them between this one analyzer/fixer, I don't know that we have a standard convention. I would just make them internal static methods in the analyzer and use them directly in the fixer.